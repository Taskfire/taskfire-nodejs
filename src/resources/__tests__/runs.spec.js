// import { nextTick } from 'promise-callbacks'
import RunResources from '../runs'
import Client from '../../client'

jest.mock('../../client')

describe('runs', () => {
  let runs
  let client

  beforeEach(() => {
    client = new Client()
    runs = new RunResources(client)
  })

  test('tasks is instance of TaskResource', () => {
    expect(runs).toBeInstanceOf(RunResources)
  })

  test('tasks has the required public API', () => {
    expect(runs.get).toEqual(expect.any(Function))
    expect(runs.list).toEqual(expect.any(Function))
  })

  describe('Basic API', () => {
    beforeEach(() => {
      runs._request = jest.fn()
    })

    test('get sends request', () => {
      runs.get('runId')
      expect(runs._request).toHaveBeenCalledTimes(1)
      expect(runs._request.mock.calls).toMatchSnapshot()
    })

    test('list sends request', () => {
      runs.list('taskId', { x: 1 })
      expect(runs._request).toHaveBeenCalledTimes(1)
      expect(runs._request.mock.calls).toMatchSnapshot()
    })
  })
})
