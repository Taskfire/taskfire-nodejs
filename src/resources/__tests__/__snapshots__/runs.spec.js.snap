// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs Basic API get sends request 1`] = `
Array [
  Array [
    Object {
      "method": "GET",
      "url": "/runs/runId",
    },
    undefined,
  ],
]
`;

exports[`runs Basic API list sends request 1`] = `
Array [
  Array [
    Object {
      "method": "GET",
      "query": "taskId",
      "url": "/runs",
    },
    Object {
      "x": 1,
    },
  ],
]
`;
